multimodal/src/autogluon/multimodal/data/infer_types.py

```diff
@@ -208,6 +215,62 @@ def is_imagepath_column(
         return False
 
 
+def is_imagebytearray_column(
+    data: pd.Series,
+    col_name: str,
+    sample_n: Optional[int] = 500,
+) -> bool:
+    """
+    Identify if a column contains valid image bytearray.
+
+    Parameters
+    ----------
+    data
+        One column of a multimodal pd.DataFrame for training.
+    col_name
+        Name of column.
+
+    Returns
+    -------
+    Whether the column is an image-bytearray column.
+    """
+    sample_num = min(len(data), sample_n)
+    data = data.sample(n=sample_num, random_state=0)
+    data = data.tolist()
+    failure_count = 0
+    for image_bytearrays in data:
+        success = False
+        if not isinstance(image_bytearrays, list):
+            image_bytearrays = [image_bytearrays]
+
+        for img_bytearray in image_bytearrays:
+            try:
+                with PIL.Image.open(BytesIO(img_bytearray)) as img:
```

Comment:
The benchmarking was actually done on a Sagemaker inference endpoint so it would have not gone through this `infer_type` procedure. This is only done during the initial phase of training, which time should be negligible compared to the training time.